### ISSUE TITLE
Computer Freeze

### ISSUE THREAD

**Issue by tkgmizzou** [thoughtnetwork/thought#3]
My computer froze while I was in the middle of downloading the wallet. I had to turn off my computer and then reboot it. I went back to Thought Core thinking I would be able to finish the download. However, when I click on Thought Core, it brings up the window, thinks about for 10 seconds and then a popup window appears telling me I need to encrypt the wallet. When I click ok the popup window goes away and the Thought Core window goes away. Tried uninstalling and reinstalling but the same issue occurs. Any help will be appreciated.

**Comment by philgrim2** [comment]
I’m assuming you’re on Windows?  Or something else?

 

Also – great help from the community on our Discord (https://discord.gg/GWKWwVK58n)

 

 

From: tkgmizzou ***@***.***> 
Sent: Tuesday, July 27, 2021 5:07 PM
To: thoughtnetwork/thought ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [thoughtnetwork/thought] Computer Freeze (#3)

 

My computer froze while I was in the middle of downloading the wallet. I had to turn off my computer and then reboot it. I went back to Thought Core thinking I would be able to finish the download. However, when I click on Thought Core, it brings up the window, thinks about for 10 seconds and then a popup window appears telling me I need to encrypt the wallet. When I click ok the popup window goes away and the Thought Core window goes away. Tried uninstalling and reinstalling but the same issue occurs. Any help will be appreciated.

**Comment by tkgmizzou** [comment]
I'm on a Mac.

**Comment by philgrim2** [comment]
So, when a node crashes (power outage, lockup, etc.) the block database can get corrupted.

Probably the easiest thing to do is to delete the blockchain and re-download.  

 

There’s a shortcut, though.  You can go to https://idea-01.insufficient-light.com, which is my quick-start guide to making nodes (it’s more Linux-oriented, but for this purpose that doesn’t matter).  Scroll down until you see some links to a tarball of the blockchain that you can download and unpack into your data directory.  Do that and then restart your wallet, and it will catch up from the date of the backup (early this morning, coincidentally).  You should be fine then.

 

The directory you’re looking for should be in /users/you/Library/Application Support/ThoughtCore.  You’ll want to untar the tarball in that folder.

The tarball contains blocks/ chainstate/ database/ and evodb/  

It will overwrite what you have there with a good version.  Should fix you right up.

**Comment by tkgmizzou** [comment]
hi phil, your explanation makes sense. i'm really sorry but while i wish i knew more about computing most anything past downloading is over my head. how do i delete the blockchain because i'm definitely willing to start over?

**Comment by philgrim2** [comment]
You can use Finder to navigate to the folder  Library/Application Support/ThoughtCore that is in your home folder.  (It might be hidden – I don’t have a Mac, so I’m not sure)

 

In that folder will be folders named blocks, chainstate, database, and evodb (the last two might not be there, depending on how far the download got).  Just delete those folders, then run the wallet again and the download will restart.

**Comment by tkgmizzou** [comment]
So I went to Library/Application Support but when I arrived there I was not given the option of clicking Thought Core?

**Comment by tkgmizzou** [comment]
Thought Core is in my Applications but not in Application Support?

**Comment by philgrim2** [comment]
The program itself is in Applications.  It saves its data in Application Support.

 

In the Application Support folder in Finder, try pressing Command + Shift + . (period)

 

If the ThoughtCore folder is hidden, this should make it appear.

**Comment by tkgmizzou** [comment]
Okay, did as you said but still don’t see Thought Core?

**Comment by tkgmizzou** [comment]
I found a file on my desktop which has the following folders:
Applications
Thought Core
.background
.fseventsd
.Trashes

?

**Comment by philgrim2** [comment]
Is there anything in the ThoughtCore folder there?

**Comment by tkgmizzou** [comment]
No, it’s the icon from before when clicked acts like it’s downloading the wallet and then the window pop up saying make sure to encrypt your wallet, etc.

**Comment by philgrim2** [comment]
Ok, let’s try this instead.

 

In Finder, click on the search button, and search for wallet.dat

That file will be in the same folder we’re looking for.

**Comment by tkgmizzou** [comment]
So I have 2 items I can see saying anything about Thought Core.

The line under Applications which when I click on it takes me to the wallet and ends up saying I need to encrypt and then goes away.

Then, on my desktop, a file box icon saying Thought-Core which when I click on it gives me the color Applications box, the color Thought Core box and three ghosted boxes as follows: .background, .fseventsd and .Trashes.

Not sure if there are others hidden but these are the ones I can see?

**Comment by philgrim2** [comment]
Was Finder able to find wallet.dat?

**Comment by tkgmizzou** [comment]
I put wallet.dat into the Finder search and nothing shows up.

**Comment by philgrim2** [comment]
I’m starting to run out of ideas here – like I said, I don’t have or use a Mac.

 

The dialog box you describe that tells you to encrypt your wallet is displayed when the software finishes creating the file wallet.dat.  That would indicate that that file should exist somewhere.  If it doesn’t, I’m not sure what’s going on.

 

If you are able to uninstall the software and try installing fresh, there are some initial dialogs that happen during installation, one of which tells you where the data directory is going to be.  You could try that.

**Comment by tkgmizzou** [comment]
Tried the following:

Deleted Thought Core and all the downloads of Thought Core from my computer. I think everything.

Turned off the computer and then turned back on.

Went to GitHub and went through the following:

Downloaded the file from thought-0.18.1/Thought-Qt.dmg.

Click on the Download and window pops up with 2 icons:

Applications
Thought Core

No words or dialogues.

Put Thought Core in Apps folder

No words or dialogues.

Go to Apps and select Thought Core.

No words or dialogues.

Thought Window pops up and says at the bottom “Loading block index”. The. window pops up saying I need to “make sure to encrypt your wallet…” and I’m back to the original issue.

I realize that all you can do is try to help but I think either I’m not deleting all instances of Thought from my comp or there’s something in the program which doesn’t enable a start-over?

Or it could be something completely different?

**Comment by philgrim2** [comment]
There is probably something in your user directory or registry that saves what your config directory is set to, that didn’t go away when you uninstalled.  I’m not sure what that is but I’ll try to find out.

**Comment by tkgmizzou** [comment]
K, thank you for trying.

**Comment by philgrim2** [comment]
Ok, look in Finder again, in your home folder /Library/Preferences.  See if there’s a file like com.thought-live.Thought-Qt.plist

**Comment by philgrim2** [comment]
I’ve been looking at the actual source code (and the documentation for QSettings, which is the class that saves your preferences, including your data directory)

If it’s either mislabeled or in the wrong place, there is something happening on your computer that I truly don’t understand.

 

I mean, at least I’m learning more about Macs from this, but I still don’t understand what’s happening on your computer.  I know people with Macs who use this software and it generally works.  I actually asked one of them for guidance a bit ago, and he found that .plist file, in his home directory, under the Library/Preferences folder, which is where the code says it should be.  If it’s not there on yours, I have no idea what has happened.
